libyul/AsmAnalysis.cpp

```diff
@@ -319,6 +319,21 @@ std::vector<YulString> AsmAnalyzer::operator()(FunctionCall const& _funCall)
 				"The underlying opcode will eventually undergo breaking changes, "
 				"and its use is not recommended."
 			);
+		else if (
+				m_evmVersion.supportsTransientStorage() &&
+				(
+					_funCall.functionName.name == "tstore"_yulstring ||
+					_funCall.functionName.name == "tload"_yulstring
+				)
+			)
+			m_errorReporter.warning(
+				2394_error,
+				nativeLocationOf(_funCall.functionName),
+				// TODO: Make warning longer and more menacing :)
+				"Please, only use transient storage if you know what you are doing. "
+				"If you know what you are doing, please don't use transient storage (unless strictly needed)."
```

Comment:
```suggestion
				"Due to a bug in the specification of EIP-1153, transient storage can break the composability of smart contracts. "
				"Since transient storage is cleared only at the end of the transaction and not at the end call frames, "
				"your contract may unintentionally misbehave when invoked multiple times in a complex transaction. "
				"To avoid this, be sure to clear all transient storage at the end of any call to your contract. "
				"The use of transient storage as reentrancy guards that are cleared at the end of the call is safe."
```

as a first suggestion for improvement.
It's also enough to emit it at ``tstore`` - technically, ``tload`` cannot cause issues.